it/pages/contactPreference/EmailPreferencePageSpec.scala

```diff
@@ -121,6 +125,52 @@ class EmailPreferencePageSpec extends BasePageISpec {
         }
       }
 
+      "the user doesn't have an email address and selects 'Yes'" should {
+
+        "redirect to the the (contact pref) Add email address page" in {
+
+          given.user.isAuthenticated
+
+          And("a successful response for an individual is stubbed")
+          VatSubscriptionStub.stubCustomerInfoNoEmail
+
+          When("'Yes' is selected")
+          val res = submit(Yes)
+
+          res should have(
+            httpStatus(Status.SEE_OTHER),
+            redirectURI(controllers.contactPreference.routes.AddEmailAddressController.show().url)
+          )
+        }
+
+        s"add the ${SessionKeys.contactPrefUpdate} key and value to session" in {
```

Comment:
I wouldn't add this as a separate test as integration tests are quite slow and it covers the same scenario above. I'd suggest either adding the `SessionCookieCrumbler.getSessionMap(result).get(SessionKeys.contactPrefUpdate) shouldBe Some("true")` check to the above test or just testing the session part in the controller unit test spec

Reply:
Removed as covered in unit tests